docs/reference/schemas/product_update_api_v3.yaml

```
@@ -0,0 +1,10 @@
+title: Product update API V3
+x-stoplight:
+  id: qnmmu2yuzc1xf
+type: object
+description: Model for creating or updating products using the v3 version of the product update API.
+properties:
+  packagings:
+    $ref: ./packagings/packagings.yaml
+  packagings_add:
+    $ref: ./packagings/packagings.yaml
```

If both fields are exclusive you can specify it with OpenAPI
I think it's:
```suggestion
  oneOf:
  - packagings:
      $ref: ./packagings/packagings.yaml
  - packagings_add:
      $ref: ./packagings/packagings.yaml
```

---

@alexgarel it cannot be done exactly like that, oneOf expects object schemas, not properties linking to object schemas.

e.g.

```
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
```

So if we wanted to do that, we would first have to define a model "packagings_packagings" that would just contain packagings: $ref: ./packagings/packagings.yaml
And then another "packagings_packagings_add" that would contain "packagings_add: $ref: ./packagings/packagings.yaml", and then we could do the oneOf between the 2.

I think it adds a lot of complexity for not much benefits. Especially since we would have to do that on all similar fields (e.g. for all taxonomies, we will have fields like "categories_add" to add a category to the existing list of categories.

---

ok.